indico_toolkit/indico_wrapper/modelop.py

```diff
@@ -0,0 +1,204 @@
+from indico import IndicoClient
+from indico.queries import GraphQLRequest
+import json
+import string
+
+
+class ModelOp:
+
+    """
+    Class to assit in Model Group Operation calls.
+    """
+
+    def __init__(self, client: IndicoClient):
+        self.client = client
+        self.task_type = None
+
+    def get_model_group(self, model_group_id: int) -> dict: 
+        """
+        Get specified model group information.
+        Args:
+            model_group_id (int): id of model group 
+        Returns:
+            dict: attributes of the specified model group id 
+        """
+
+        query = """
+        
+query GetModelGroup($id: Int) {
+  modelGroups(modelGroupIds: [$id]) {
+    modelGroups {
+      id
+      name
+      status
+      taskType
+      selectedModel {
+        id
+        status
+      }
+      modelOptions {
+        id
+        samplingStrategy
+        modelTrainingOptions
+      }
+    }
+  }
+}
+                """
+        
+        results = self.client.call(
+            GraphQLRequest(query, {"modelGroupId": model_group_id})
+        )
+        models = results["modelGroups"]["modelGroups"]
+        for model in models:
+            if model["id"] == model_group_id:
+                #assign task type for advanced model options
+                self.task_type = model["taskType"]
+                #assign model settings for validation
+                self.model_setings = json.loads(model["modelOptions"]["modelTrainingOptions"])
+                return model
+
+    def retrain_model(self, model_group_id: int):
+        """
+        Forces model retrain, used within update_model_settings to apply new Training Options.
+        Args:
+            model_group_id (int): id of model group needing retraining
+        """
+
+        query = """
+
+mutation retrainMoonbowModelGroup($modelGroupId: Int!) {
+  retrainModelGroup(modelGroupId: $modelGroupId, forceRetrain: true) {
+    id
+    status
+    __typename
+  }
+}    
+                """
+        
+        self.client.call(GraphQLRequest(query, {"modelGroupId": model_group_id}))
+        return
+
+    def list_models(self, model_group_id: int) -> list[dict]:
+        """
+        Returns all avaible models within a model group. Used to show new model ID after retraining. 
+        Args:
+            model_group_id (int): id of model group 
+        Returns:
+            list[dict]: List of dictionaries for each model and its corresponding attributes
+        """
+
+        query = """
+
+query getMoonbowModelGroupModels($modelGroupId: Int!) {
+  modelGroup(modelGroupId: $modelGroupId) {
+    id
+    models {
+      id
+      modelOptions
+    }
+  }
+}
+                """
+        
+        results = self.client.call(
+            GraphQLRequest(query, {"modelGroupId": model_group_id})
+        )
+        models = results["modelGroup"]["models"]
+        self.latest_model = models[0]
```

Comment:
is the first model in that list always the most recent? how is this different from your `get_model_group` method? (what aren't you getting in this?)